plugins/inputs/snmp/snmp.go

```diff
@@ -752,6 +760,14 @@ func fieldConvert(conv string, v interface{}) (interface{}, error) {
 		return v, nil
 	}
 
+	if conv == "enum" {
+		return tr.SnmpFormatEnum(ent.Name, ent.Value, false)
+	}
+
+	if conv == "enum(1)" {
+		return tr.SnmpFormatEnum(ent.Name, ent.Value, true)
+	}
```

Comment:
This whole thing should really be a `switch` statement, but I don't want you to change it in this PR. So please ignore this comment for this PR.

Reply:
I can do it if you want, but what about that one case which is doing `scanf`?

Reply:
Don't do it here. But we really should think about converting this in another PR. The exceptions can be handled in default or in if cases after/before the switch statement...